numba/tests/test_fancy_indexing.py

```diff
@@ -301,5 +301,188 @@ def np_new_axis_setitem(a, idx, item):
+        # contiguity as NumPy
+        self.assertEqual(got.shape, expected.shape)
+        self.assertEqual(got.dtype, expected.dtype)
+        np.testing.assert_equal(got, expected)
+
+        # Check a copy was *really* returned by Numba
+        self.assertFalse(np.may_share_memory(got, expected))
+
+    def check_setitem_indices(self, arr_shape, index):
+        @njit
+        def set_item(array, idx, item):
+            array[idx] = item
+
+        arr = np.random.randint(0, 11, size=arr_shape)
+        src = arr[index]
+        expected = np.zeros_like(arr)
+        got = np.zeros_like(arr)
+
+        set_item.py_func(expected, index, src)
+        set_item(got, index, src)
+
+        # Note: Numba may not return the same array strides and
+        # contiguity as NumPy
+        self.assertEqual(got.shape, expected.shape)
+        self.assertEqual(got.dtype, expected.dtype)
+
+        np.testing.assert_equal(got, expected)
+
+    def test_getitem(self):
+        # Cases with a combination of integers + other objects
+        indices = self.indexing_cases.copy()
+
+        # Cases with permutations of either integers or objects
+        indices += self.generate_random_indices()
+
+        for idx in indices:
+            with self.subTest(idx=idx):
+                self.check_getitem_indices(self.shape, idx)
+
+    def test_setitem(self):
+        # Cases with a combination of integers + other objects
+        indices = self.indexing_cases.copy()
+
+        # Cases with permutations of either integers or objects
+        indices += self.generate_random_indices()
+
+        for idx in indices:
+            with self.subTest(idx=idx):
+                self.check_setitem_indices(self.shape, idx)
+
+    def test_unsupported_condition_exceptions(self):
+        err_idx_cases = [
+            # Cases with multi-dimensional indexing array
+            ('Multi-dimensional indices are not supported.',
+             (0, 3, np.array([[1, 2], [2, 3]]))),
+            # Cases with more than one indexing array
+            ('Using more than one non-scalar array index is unsupported.',
+             (0, 3, np.array([1, 2]), np.array([1, 2]))),
+            # Cases with more than one indexing subspace
+            # (The subspaces here are separated by slice(None))
+            ("Using more than one indexing subspace is unsupported." + \
+             " An indexing subspace is a group of one or more consecutive" + \
+             " indices comprising integer or array types.",
+             (0, np.array([1, 2]), slice(None), 3, 4))
+        ]
+        
+        for err, idx in err_idx_cases:
+            with self.assertRaises(TypingError) as raises:
+                self.check_getitem_indices(self.shape, idx)
+            self.assertIn(
+                err,
+                str(raises.exception)
+            )
+
+
+# if __name__ == '__main__':
+#     unittest.main()
+x = TestFancyIndexingMultiDim()
+x.test_getitem()
```

Comment:
This seems to be an unrelated change?

Reply:
Resolved in [233b1d1](https://github.com/numba/numba/pull/8238/commits/233b1d1b0d2a40c56083cdbfe8e62c9fede67aaa)